x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/inference/trainedmodel/tree/TreeNode.java

```diff
@@ -190,7 +196,18 @@ public void writeTo(StreamOutput out) throws IOException {
         out.writeInt(splitFeature);
         out.writeDouble(splitGain);
         out.writeVInt(nodeIndex);
-        out.writeDouble(leafValue);
+        if (out.getVersion().onOrAfter(Version.V_7_7_0)) {
+            out.writeDoubleArray(leafValue);
+        } else {
+            if (leafValue.length > 1) {
+                throw new IOException("[leaf_value] only supports multiple values after version 7.7.0");
```

Comment:
I am not sure about this.  To what extent does it cause a problem for the receiving node?  Could it cause problems for the reading of a subsequent wire transfer?  I don't know the serialisation code well enough to say whether this is OK or not.

Also, would it make sense to an end user who saw it?  I guess they'd see a remote transport exception from a 7.6 coordinating node wrapping this exception.  But which endpoint would they have called?  Would `leaf_value` make sense to somebody who had called that endpoint?

Finally, maybe we can defend against this at an earlier stage by banning creation of multi-class classification jobs until all nodes in the cluster are on version 7.7.

Reply:
> To what extent does it cause a problem for the receiving node?

Receiver node gets an exception. The serialization fails. 

This type of thing is done in aggregations as well https://github.com/elastic/elasticsearch/blob/master/server/src/main/java/org/elasticsearch/search/aggregations/bucket/composite/CompositeValuesSourceParserHelper.java#L72

I think throwing here is OK.

As for the message, I will fix it. I think something like `Multi-class classification model attempted to serialize to node before version X. Multi-class classification is only supported if the node is greater than version X`

> maybe we can defend against this at an earlier stage by banning creation of multi-class classification jobs until all nodes in the cluster are on version 7.7.

I agree, right now it is not possible to create a multi-class model via the analytics process. Once it is, we should defend against it there. 

Additionally, on `PUT` I can verify the minimum node version